CONTRIBUTING.md

```diff
@@ -65,19 +65,52 @@ Thus:
 - if the list includes `next` but no release tags: the fix/enhancement will come in the nearest minor release.
 - if the list includes `main` but no release tags: the fix/enhancement will come in the nearest patch release.
 
+## Proposing new changes
+
+If you want to add functionality and features that don't fall under an existing and accepted issue please add a [Kubernetes Enhancement Proposal (KEP)][keps] in a pull request to the `railgun/keps/` directory.
+
+To the best of your ability: take some time before you propose something new to search through some of the existing KEPs to make sure that your proposal isn't a duplicate nor is it colliding with anything others.
+
+Read our [KEP Documentation][kepstart] which provides some guidelines about how to get started and contributing to new or existing KEPs.
+
+[keps]:https://github.com/kubernetes/enhancements/blob/master/keps/README.md
+[kepstart]:https://github.com/Kong/kubernetes-ingress-controller/blob/next/railgun/keps/README.md
+
```

Comment:
Before we accept this as a community guideline I'd like us (the maintainers) to consider the opportunity cost of using KEPs.

Basically I'd like us to convince ourselves that over the course of the last months the KEP process (and the associated PR burden, and the choice of tooling) lifts its own weight and we can commit to maintaining the KEPs in good health (our process is centered around other things, not KEPs today)

Suggestion: split this out to a separate PR and merge the non-controversial bits of this PR.

Reply:
I did not realize this was contentious. Removed in 65a03b330d0f868c93a85c62cc765651dbd78131

Reply:
Agreed for community stuff. Currently most feature requests/proposals we get from the community are fairly small in scope and can be described and managed well enough informally.